js/zap-common.js

```diff
@@ -0,0 +1,64 @@
+/*eslint-disable no-unused-vars*/
+function getParameterByName(url, name) {
+    name = name.replace(/[\[]/, "\\[").replace(/[\]]/, "\\]");
+    var regex = new RegExp("[\\?&]" + name + "=([^&#]*)"),
+        results = regex.exec(url);
+    return results === null ? "" : decodeURIComponent(results[1].replace(/\+/g, " "));
+}
+
+function getSearchResults(query, youTubeDataApiKey, onData, onFail) {
+    $.getJSON("https://www.googleapis.com/youtube/v3/search", {
+        key: youTubeDataApiKey,
+        part: "snippet",
+        q: query,
+        type: "video"
+    }, onData).fail(onFail);
+}
+
+function getAutocompleteSuggestions(query, callback) {
+    $.getJSON("https://suggestqueries.google.com/complete/search?callback=?", {
```

Comment:
@shakeelmohamed To use jsonp for cross-domain request. Without it, we'll see origin not allowed err.